internal/fnruntime/container.go

```diff
@@ -157,27 +157,18 @@ func newContainerEnvFromStringSlice(envStr []string) *runtimeutil.ContainerEnv {
 // prepareImage will check local images and pull it if it doesn't
 // exist.
 func (f *ContainerFn) prepareImage() error {
-	// check image existence
-	args := []string{"image", "ls", f.Image}
-	cmd := exec.Command(dockerBin, args...)
-	var output []byte
-	var err error
-	if output, err = cmd.CombinedOutput(); err != nil {
-		return fmt.Errorf("failed to check local function image %q: %w", f.Image, err)
-	}
-	if strings.Contains(string(output), strings.Split(f.Image, ":")[0]) {
-		// image exists locally
-		return nil
-	}
-	args = []string{"image", "pull", f.Image}
+	// We always try to pull the image regardless if the tag has been seen in
+	// the local cache. Because we want to ensure we have the latest release for
+	// tags like v1 and v1.2 that are moving tags.
+	args := []string{"image", "pull", f.Image}
```

Comment:
Will this work for local containers that are not in a remote registry? If I build `my-func` locally and then try to test it `docker pull my-func` will fail and prevent running the function at line 172?

Reply:
Good point! We can make fetching image a best-effort based operation.
If there is an error (e.g. the image doesn't exist in the remote registry), we simply ignore the error from the image pulling operation instead of failing the whole pipeline.

Reply:
That's a good catch. It shouldn't fail for local functions.

Another thing I forgot to ask, will this negate the caching of container images ? Say, I have `my-fn:v1.1.1` on my machine and no new version of `my-fn` has been released after `v1.1.1`, will it just check do a lightweight check for newer version or will it blindly refetch `v1.1.1` ?

Reply:
Updated.

> Another thing I forgot to ask, will this negate the caching of container images ?

No. It only fetches the missing layer(s).

Reply:
ok.